src/main/resources/page/js/header.js

```diff
@@ -0,0 +1,84 @@
+class NavBarItems {
+    constructor(singleNavBarItems, dropDownNavBarItems) {
+        this.singleNavBarItems = singleNavBarItems;
+        this.dropDownNavBarItems = dropDownNavBarItems;
+        this.addSingleItems(singleNavBarItems);
+        this.addDropDownItems(dropDownNavBarItems);
+    }
+
+    addSingleItems(singleNavBarItems) {
+        for (let [linkName, href] of Object.entries(singleNavBarItems)) {
+            let singleNavItem = this.createSingleItems(linkName, href);
+            document.getElementById('navbar-items').append(singleNavItem);
+        }
+    }
+
+    createSingleItems(linkName, href) {
+        let singleNavItem = document.createElement("li");
+        singleNavItem.classList.add("nav-item");
+        let link = document.createElement("a");
+        link.classList.add("nav-link", "active");
+        link.href = href;
+        link.innerHTML = linkName;
+        link.target = '_blank';
+        singleNavItem.append(link);
+        link.onclick = function () {
+            console.log('You clicked to ' + linkName + '!');
+        }
+        return singleNavItem;
+    }
+
+    addDropDownItems(dropDownNavBarItems) {
+        for (let [dropDownName, links] of Object.entries(dropDownNavBarItems)) {
+            let dropDownLink = this.createDropDownLink(dropDownName);
+            let dropDownMenu = this.createDropDownMenu();
+            dropDownLink.append(dropDownMenu);
+            document.getElementById("navbar-items").append(dropDownLink);
+            for (let [linkName, href] of Object.entries(links)) {
+                let dropDownItems = this.createDropDownItems(linkName, href);
+                dropDownMenu.append(dropDownItems);
+            }
+        }
+    }
+
+    createDropDownLink(dropDownName) {
+        let dropDownLinkContainer = document.createElement("li");
+        dropDownLinkContainer.classList.add("nav-item", "dropdown");
+        let dropDownLink = document.createElement("a");
+        dropDownLink.classList.add("nav-link", "dropdown-toggle");
+        dropDownLink.id = "navbarDropdownMenuLink";
+        dropDownLink.setAttribute("role", "button");
+        dropDownLink.setAttribute("data-bs-toggle", "dropdown");
+        dropDownLink.setAttribute("aria-expanded", "false");
+        dropDownLink.innerHTML = dropDownName;
+        dropDownLinkContainer.append(dropDownLink);
+        return dropDownLinkContainer;
+    }
+
+    createDropDownItems(linkName, href, parent) {
+        let dropDownItems = this.createDropDownItems(linkName, href);
```

Comment:
We do not need to use a local variable in this case.

Reply:
I can see, I will delete it, thanks for the good insight!